src/arraytraits.rs

```diff
@@ -94,6 +94,46 @@ impl<S, D, I> IndexMut<I> for ArrayBase<S, D>
     }
 }
 
+impl<'a, A, D> ArrayView<'a, A, D>
+where
+    D: Dimension,
+{
+    /// Get a reference of a element through the view.
+    ///
+    /// This is a replacement of `Index::index` since it forces us a wrong lifetime
+    /// https://github.com/bluss/rust-ndarray/issues/371
+    pub fn elem_ref<I: NdIndex<D>>(&self, index: I) -> &'a A {
```

Comment:
I think this is a good idea. We should make something that closely aligns with the design of the methods for element access that we have, I'm thinking of the Index trait and of the methods `.get()` and `.uget()`.

I'd like to find a good name for this -- something that's similar to `get` or `index`. Also, we probably want an unchecked version of this method too (like `uget`).